addons/website_sale_digital/controllers/main.py

```diff
@@ -86,7 +86,10 @@ def download_attachment(self, attachment_id):
         res_id = attachment['res_id']
         purchased_products = request.env['account.move.line'].get_digital_purchases()
 
-        if res_model == 'product.product':
+        if request.env['ir.attachment'].search([('id', '=', int(attachment_id))]):
```

Comment:
I, too, find it confusing to skip the redirects with a `if: pass`.

Also, it's important that the security check is obvious to the reader. Here, it's not immediate that `.sudo()` is omitted on purpose from the `search` call while it was necessary with the `search_read` call. Instead, could you please use `check_access_{rights,rules}` to make the check obvious?